app/controllers/ops_controller/settings/label_tag_mapping.rb

```diff
@@ -128,8 +128,21 @@ def lt_map_get_form_vars
   end
 
   def label_tag_mapping_add(entity, label_name, cat_description)
+    entity_str = ''
     prefix = ContainerLabelTagMapping::AUTOTAG_PREFIX
-    entity_str = entity.nil? ? "" : entity.underscore
+
+    # The entity is a string in the form "Provider::ResourceType".
+    if entity
+      prefix, entity = entity.split('::')
+      # Backwards compatibility
+      if prefix && entity
+        prefix.downcase!
+        entity_str = entity.underscore
+      else
+        prefix = ContainerLabelTagMapping::AUTOTAG_PREFIX
+      end
+    end
```

Comment:
This method was already too long, please extract a method to compute category name (or some other split).